doc/build-helpers/fetchers.chapter.md

```diff
@@ -157,27 +157,383 @@ Here are security considerations for this scenario:
 
   In more concrete terms, if you use any other hash, the [`--insecure` flag](https://curl.se/docs/manpage.html#-k) will be passed to the underlying call to `curl` when downloading content.
 
-## `fetchurl` and `fetchzip` {#fetchurl}
+[]{#fetchurl}
+## `fetchurl` {#sec-pkgs-fetchers-fetchurl}
 
-Two basic fetchers are `fetchurl` and `fetchzip`. Both of these have two required arguments, a URL and a hash. The hash is typically `hash`, although many more hash algorithms are supported. Nixpkgs contributors are currently recommended to use `hash`. This hash will be used by Nix to identify your source. A typical usage of `fetchurl` is provided below.
+`fetchurl` returns a derivation which downloads content from a given URL and stores the unaltered contents within the Nix store.
+
+It uses {manpage}`curl(1)` internally, and allows its behaviour to be modified by specifying a few attributes in the argument to `fetchurl` (see the documentation for attributes `curlOpts`, `curlOptsList`, and `netrcPhase`).
+
+The resulting [store path](https://nixos.org/manual/nix/stable/store/store-path) is determined by the hash given to `fetchurl`, and also the `name` (or `pname` and `version`) values.
+
+If neither `name` nor `pname` and `version` are specified when calling `fetchurl`, it will default to using the [basename](https://nixos.org/manual/nix/stable/language/builtins.html#builtins-baseNameOf) of `url` or the first element of `urls`.
+If `pname` and `version` are specified, `fetchurl` will use those values and will ignore `name`, even if it is also specified.
+
+### Inputs {#sec-pkgs-fetchers-fetchurl-inputs}
+
+`fetchurl` expects a single argument with the following attributes:
```

Comment:
It's a copy-paste from one of my first PRs where I updated the content in the manual :)
There are a bunch of other places in the manual to update the wording, but I'll do that in a follow-up PR to keep this one focused.

Reply:
I'm keeping the "attributes" bit to avoid confusion (e.g. someone thinking 'are attribute set "entries" different from their attributes?').

```suggestion
`fetchurl` requires an attribute set with the following attributes:
```